### DISCUSSION TITLE
Feature Request: Add `key`-prop to `MenuItem`

### DISCUSSION THREAD

**Discussion by nimo23** [primefaces/community#3106] [PrimeReact] [answered]
Please add the prop `key` to the `MenuItem` so that react can render based on the `key`:

```
/**
 * Defines model of MenuItem API.
 * @group Model
 */
export interface MenuItem {
   // TODO add key prop:
   /**
     * Unique key of the node.
     */
    key?: string | number | undefined;

    /**
     * Unique identifier of the menuitem.
     */
    id?: string | undefined;
    /**
     * Text of the menuitem.
     */
    label?: string | undefined;

..
```

**Comment by melloware** [comment]
Menu's should already be rendering by a unique key the menu creates.  Are you not seeing that or is there a bug?

**Reply by nimo23** [reply]
I can't see PrimeReact automatically generating a unique key for a menu item. By the way, I usually expect the `key`-prop to be assigned by the user (since he knows which `key` to choose). I expect something like this:

```
const userNode: MenuItem = {
	id: user.id,
	key: user.id, // cannot set key as Interface MenuItem does not have a key prop
	label: user.username,
	expanded: true,
	data: {
	    name: user.username,
	},
	items: []
};
```

**Reply by melloware** [reply, marked as answer]
But the key is just used by react for uniqueness and the component gaurantees that.  What are you using key for?  That is what the `data` element of the menu is for so you can add any metadata you want.
